CMakeLists.txt

```diff
@@ -127,32 +125,26 @@ if(openmp)
   find_package(OpenMP)
   if(OPENMP_FOUND)
     # In CMake 3.9+, can use the OpenMP::OpenMP_CXX imported target
-    list(APPEND cxxflags ${OpenMP_CXX_FLAGS})
+    add_compile_options(${OpenMP_CXX_FLAGS})
     list(APPEND ldflags ${OpenMP_CXX_FLAGS})
   endif()
 endif()
 
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
-list(APPEND cxxflags -O2)
-if(debug)
-  list(REMOVE_ITEM cxxflags -O2)
-  list(APPEND cxxflags -g -O0)
+if (NOT CMAKE_BUILD_TYPE)
+    add_compile_options(-O2)
```

Comment:
Perhaps rather than setting this default -O2, we ought to just set `CMAKE_BUILD_TYPE` to `Release` if it's not explicitly set. Note that you'll have to set it as a [cache variable](https://cmake.org/cmake/help/latest/command/set.html#set-cache-entry) to get it to persist.

Reply:
I might not have been setting this right, but I got it to work with an if-statement and no cache.